docs/source/upgrade.rst

```diff
@@ -0,0 +1,759 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, :class:`GradientMethod` clears the gradient before running backprop by :meth:`Link.cleargrads`.
+It means that the gradient of each parameter is initialized by ``None`` instead of a zero array.
+
+In most cases, you do not need to update your code.
+If your code relies on the zeroing initialization, you have to fix your code to explicitly initialize the gradient by zero, or to pass ``False`` to :meth:`GradientMethod.use_cleargrads`.
+
+.. _upgrade-update-rule:
+
+GradientMethod is redesigned to allow parameter-specific update rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, the new class :class:`UpdateRule` is used to define an update rule specific to each :class:`Parameter` object.
+The :class:`UpdateRule` is set to each :class:`Parameter` object, and is used at each update step.
+This object implements *an update formula* using the data and gradient arrays.
+
+Each :class:`UpdateRule` object can also hold its own hook functions similar to :class:`Optimizer`.
+The built-in hook functions except for :class:`~optimizer.GradientClipping` can also be used as a hook function of :class:`UpdateRule`.
+
+**In most cases, you do not have to update your codes** because each optimizer automatically sets up an appropriate :class:`UpdaterRule` object to each parameter.
+
+**If you are using a custom gradient-based optimizer implementation, you need to update the implementation.**
+The following list shows what you have to do.
+
+- Write a subclass of :class:`UpdateRule` that implements the update rule.
+- Rewrite your :class:`GradientMethod` implementation.
+  The new implementation only has to set up the update rule for each parameter in the target link.
+
+You can see live examples in `the optimizer implementations provided by Chainer <https://github.com/pfnet/chainer/tree/master/chainer/optimizers>`_.
+
+
+Serializer
+----------
+
+None is serializable
+~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, all serializers start supporting ``None`` value to be serialized and deserialized.
+Users' code can rely on this feature, i.e., it can serialize and deserialize ``None`` value with any given serializer.
+This change only affects your code if it provides its own serializer implementations.
+
+
+Trainer and Extension
+---------------------
+
+.. _upgrade-pass-raw-arrays-to-loss:
+
+Updater and Evaluator pass raw data arrays to the loss function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, :class:`~training.Updater` and :class:`~training.extensions.Evaluator` pass raw data arrays to the loss function without wrapping them with :class:`Variable`.
+**You might need to update your code so that the loss function (in most cases, the model's** ``__call__``, **accepts raw arrays.**
+Note that raw arrays can be directly passed to any :class:`Function`; they are automatically wrapped by :class:`Variable`.
```

Comment:
It is more kind to write explicitly what is inferred from this fact. For example, "Therefore, if the inputs to your loss function are only fed :class:`Function`, no changes are needed.